### ISSUE TITLE
Net Investment Income Tax (NIIT) logic

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/Tax-Calculator#873]
After reviewing the NIIT logic, I have some questions.  In a development branch I'm working on, the calculation of the NIIT is as follows:
```
def NetInvIncTax(e00300, e00600, e02000, e26270, c01000,
                 c00100, _feided, NIIT_thd, MARS, NIIT_trt, NIIT):
    """
    NetInvIncTax function computes Net Investment Income Tax amount
    (assume all annuity income is excluded from net investment income)
    """
    modAGI = c00100 + _feided
    NII = max(0., e00300 + e00600 + max(0., e02000 - e26270) + max(0., c01000))
    NIIT = NIIT_trt * min(NII, max(0., modAGI - NIIT_thd[MARS - 1]))
    return NIIT
```
The above code produces exactly the same results as the code on the master branch.

I have two questions about the above calculation of net investment income (NII):

(1) Why is it that `e02000 - e26270`, which is Schedule E income less partnership and S corporation net income, cannot be negative in the calculation of NII?  Looking at Form 8960 and its instructions, I haven't found anything to indicate that a Schedule E loss cannot be used to offset other NII.  Have I missed something in the thirty pages of instructions?

(2) Why is it that `c01000`, which capital gains limited to a loss of at most $3000 per Schedule D, cannot be negative in the calculation of NII?  Looking at Form 8960 and its instructions, I haven't found anything to indicate that a combined short-term and long-term capital loss (as limited by Schedule D) cannot be used to offset other NII.  Again, have I missed something?

@MattHJensen @feenberg @Amy-Xu @GoFroggyRun @zrisher @codykallen

**Comment by feenberg** [comment]
On Wed, 24 Aug 2016, Martin Holmer wrote:

> 
> After reviewing the NIIT logic, I have some questions. In a development branch I'm
> working on, the calculation of the NIIT is as follows:
> 
> def NetInvIncTax(e00300, e00600, e02000, e26270, c01000,
>                  c00100, _feided, NIIT_thd, MARS, NIIT_trt, NIIT):
>     """
>     NetInvIncTax function computes Net Investment Income Tax amount
>     (assume all annuity income is excluded from net investment income)
>     """
>     modAGI = c00100 + _feided
>     NII = e00300 + e00600 + max(0., e02000 - e26270) + max(0., c01000)
>     NIIT = NIIT_trt * min(NII, max(0., modAGI - NIIT_thd[MARS - 1]))
>     return NIIT
> 
> The above code produces exactly the same results as the code on the master branch.
> 
> I have two questions about the above calculation of net investment income (NII):
> 
> (1) Why is it that e02000 - e26270, which is Schedule E income less partnership and S
> corporation net income, cannot be negative in the calculation of NII? Looking at Form
> 8960 and its instructions, I haven't found anything to indicate that a Schedule E loss
> cannot be used to offset other NII. Have I missed something in the thirty pages of
> instructions?

In the sas calculator there is also a zero constraint on the Medicare tax 
on unearned income.

c85190 = e00300+e00600+max(0,c01000)+max(0,e02000-e26270-e85180+e85070+e85080);


> 
> (2) Why is it that c01000, which capital gains limited to a loss of at most $3000 per
> Schedule D, cannot be negative in the calculation of NII? Looking at Form 8960 and its
> instructions, I haven't found anything to indicate that a combined short-term and
> long-term capital loss (as limited by Schedule D) cannot be used to offset other NII.
> Again, have I missed something?


This is same. The sas code is probably the source of this notion.

Because this part of the code has never been compared to actual returns, 
it hasn't been tested, and is probably wrong.


dan

**Comment by martinholmer** [comment]
Dan @feenberg said:
> The sas code [taxcalc.sas] is probably the source of this notion.
> Because this part of the code [that calculates NIIT] has never been compared to actual returns, it hasn't been tested, and is probably wrong.

There is some other evidence that is consistent with your notion that the current logic is wrong.
Here is how I'm considering changing the code:
```
NII = max(0., e00300 + e00600 + (e02000 - e26270) + c01000)
```
Does that make sense?

If this change is made, the validation tests against Internet TAXSIM will fail because Internet TAXSIM appears to have the same NIIT logic as in the taxcalc.sas program.

@MattHJensen @feenberg @Amy-Xu @GoFroggyRun @zrisher @codykallen

**Comment by GoFroggyRun** [comment]
@martinholmer:

After looking at the instructions for Form 8960, I do not see anything to indicate that `e02000 - e26270` or `c01000` should be non-negative in NIIT's calculation as well.

And thus the follow code-line change
```
NII = max(0., e00300 + e00600 + (e02000 - e26270) + c01000)
```
makes sense to me.

**Comment by martinholmer** [comment]
Sean Wang (@GoFroggyRun), Thanks for your feedback on NIIT logic.

**Comment by martinholmer** [comment]
Issue #873 has been resolved by pull request #874.
